### ISSUE TITLE
When the OnColor property of Switch is set, it is applied even if when Off on iOS / Mac

### ISSUE THREAD

**Issue by muak** [dotnet/maui#15074]
### Description

When the OnColor property of Switch is set, it is applied even if when Off.
If it should be Off, it should be a light gray color.

https://github.com/dotnet/maui/assets/16084475/28bc451a-e50a-4fbc-94b8-699ec6346d8c

### Steps to Reproduce

1. Set OnColor property to Switch
2. Run the app.
3. The switch turn off.

### Link to public reproduction project repository

https://github.com/muak/SwitchIssue

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, macOS

### Affected platform versions

iOS14.7.1 / MacOS Ventura 13.3.1

### Did you find any workaround?

It will work fine if the OnColor property is not set.

**Comment by muak** [comment]
@jfversluis 
Yes, It works fine on Xamarin.Forms.

**Comment by jfversluis** [comment]
@muak we think this might be already fixed in an upcoming version @tj-devel709 is double-checking!

**Comment by muak** [comment]
@jfversluis @tj-devel709 
Thanks.
Is there any plan for this fix to be reflected in net7?

**Comment by microspaze** [comment]
> @jfversluis @tj-devel709 Thanks. Is there any plan for this fix to be reflected in net7?

@muak  You can write a custom renderer for temp fixing.

```
    public static class MauiProgram
    {
        public static MauiApp CreateMauiApp()
        {
            var builder = MauiApp.CreateBuilder();
            builder
	            .UseMauiApp<App>()
	            .ConfigureMauiHandlers(handlers =>
	            {
        if IOS
		        handlers.AddHandler(typeof(Switch), typeof(SwitchRenderer));
        #endif
	            });

	    return builder.Build();
        }
    }
   
    public class SwitchRenderer : SwitchHandler
    {
        private static readonly UIColor DefaultOffColor = UIColor.FromRGBA(120, 120, 128, 40);
        private Switch? Element => VirtualView as Switch;
        private UISwitch Control => PlatformView;

        protected override void ConnectHandler(UISwitch platformView)
        {
            base.ConnectHandler(platformView);
            if (Element == null) return;
            Element.PropertyChanged += OnElementPropertyChanged;
        }

        protected override void DisconnectHandler(UISwitch platformView)
        {
            base.DisconnectHandler(platformView);
            if (Element == null) return;
            Element.PropertyChanged -= OnElementPropertyChanged;
        }

        private void OnElementPropertyChanged(object? sender, PropertyChangedEventArgs e)
        {
            if (Element == null) return;
            var propertyName = e.PropertyName;
            if (propertyName == "IsToggled" && !Element.IsToggled)
            {
                var trackView = GetTrackSubview(Control);
                if (trackView != null)
                {
                    trackView.BackgroundColor = DefaultOffColor;
                }
            }
        }
        
        private static UIView? GetTrackSubview(UISwitch uISwitch)
        {
            if (OperatingSystem.IsIOSVersionAtLeast(13) || OperatingSystem.IsTvOSVersionAtLeast(13))
                return uISwitch.Subviews?.FirstOrDefault()?.Subviews?.FirstOrDefault();
            else
                return uISwitch.Subviews?.FirstOrDefault()?.Subviews?.FirstOrDefault()?.Subviews?.FirstOrDefault();
        }
    }
```

**Comment by returnZro** [comment]
Hi @jfversluis , is it possible to back port this to `net7`? I know there is a handler workaround mentioned above but when you got several apps targetting net7, it is time consuming to add this workaround in every app. 

`Switch` is one of the basic control and i think it is fair to expect it to work properly out of the box.

**Comment by jfversluis** [comment]
Looks like this was fixed so we can close it! 

I will suggest a backport @returnZro but not guarantees as .NET 8 is already pretty close and a workaround is already available

**Comment by jfversluis** [comment]
Fixed by #10758
